aten/src/ATen/core/Tensor.cpp

```diff
@@ -13,18 +13,15 @@ void Tensor::enforce_invariants() {
   // supported by ATen
   scalar_type();
   if (defined()) {
-    // If it's a variable - we definitely not in C2 land
-    if (!is_variable()) {
-      AT_ASSERTM(
-          impl_->dtype_initialized(),
-          "Partially-initialized tensor not supported by at::Tensor");
-      AT_ASSERTM(
-          !impl_->is_sparse(),
-          "Sparse Tensors are supported by at::Tensor, but invariant checking isn't implemented.  Please file a bug.");
-      AT_ASSERTM(
-          impl_->storage_initialized(),
-          "Partially-initialized tensor not supported by at::Tensor");
-    }
+    AT_ASSERTM(
```

Comment:
nit: TORCH_CHECK?

Reply:
I plead preexisting code XD

Reply:
I think AT_ASSERT is actually correct here (ok, you could probably argue about sparse).  But the other use cases are weird enough that it's not really considered part of the public API.